spec/posts_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:author) { User.create(name: 'David', posts_counter: 0) }
+
+  subject do
+    Post.create(author:, title: 'My first post', text: 'Here is my post.', comments_counter: 0, likes_counter: 0)
+  end
+
+  let(:first_comment) do
+    Comment.create(post: subject, author:, text: 'first comment', created_at: 'Fri, 28 Jul 2023 04:51:46')
+  end
+  let(:second_comment) do
+    Comment.create(post: subject, author:, text: 'second comment', created_at: 'Fri, 28 Jul 2023 05:51:46')
+  end
+  let(:third_comment) do
+    Comment.create(post: subject, author:, text: 'third comment', created_at: 'Fri, 28 Jul 2023 06:51:46')
+  end
+  let(:fourth_comment) do
+    Comment.create(post: subject, author:, text: 'fourth comment', created_at: 'Fri, 28 Jul 2023 07:51:46')
+  end
+  let(:fifth_comment) do
+    Comment.create(post: subject, author:, text: 'fifth comment', created_at: 'Fri, 28 Jul 2023 08:51:46')
+  end
+  let(:sixth_comment) do
+    Comment.create(post: subject, author:, text: 'sixth comment', created_at: 'Fri, 28 Jul 2023 09:51:46')
+  end
+
+  let(:first_like) { Like.create(post: subject, author:) }
+  let(:second_like) { Like.create(post: subject, author:) }
+
+  shared_examples 'a non-negative integer' do |attribute|
+    it 'counter should not be negative number' do
+      subject.send("#{attribute}=", -1)
+      expect(subject).to_not be_valid
+    end
+
+    it 'counter should be greater than or equal to zero' do
+      subject.send("#{attribute}=", 0)
+      expect(subject).to be_valid
+
+      subject.send("#{attribute}=", 1)
+      expect(subject).to be_valid
+    end
+
+    it 'counter should integer' do
+      subject.send("#{attribute}=", 1.5)
+      expect(subject).to_not be_valid
+    end
+  end
+
+  context 'check the length of the character' do
+    it 'should not be not be more than 250 character' do
+      subject.title = 'a' * 250
+      expect(subject).to be_valid
+
+      subject.title = 'a' * 251
+      expect(subject).to_not be_valid
+    end
+
+    it 'should not be empty' do
+      subject.title = nil
+      expect(subject).to_not be_valid
+    end
+
+    it_behaves_like 'a non-negative integer', :comments_counter
+    it_behaves_like 'a non-negative integer', :likes_counter
+  end
```

Comment:
- [ ] Nice job 👏 . However, you're meant to add unit spec for validations for all of  `User`, `Post`, `Comment`, and `Like` models. Kindly do so for all the files not just for post 

### Sample:
```
  describe 'validations' do
    it { should validate_presence_of(:title) }
    it { should validate_length_of(:title).is_at_most(250) }
     # Add other validations as needed for other attributes
   end
 ```
 
 ### Instruction:
![image](https://github.com/abiola-arobieke/blog-app/assets/50990925/4895c825-5b81-4575-8ec9-2303f03f821a)